target_chains/solana/cli/src/main.rs

```diff
@@ -37,90 +59,208 @@ use {
+                                    payer.pubkey(),
+                                    signature_set_keypair.pubkey(),
+                                    post_vaa_data,
+                                )?],
+                                &vec![&payer],
+                            )?;
+
+                            rpc_client.get_account_data(&vaa_pubkey).ok()
+                        }
+                    };
+
+                    println!("Verifying updates using PostedVAA account");
+
+                    let posted_vaa_data =
+                        AnchorVaa::try_deserialize(&mut vaa_account.unwrap().as_slice())?;
+                    let wormhole_message =
+                        WormholeMessage::try_from_bytes(&posted_vaa_data.payload)?;
+                    println!("wormhole_message: {wormhole_message:?}");
+                    let root: MerkleRoot<Keccak160> =
+                        MerkleRoot::new(match wormhole_message.payload {
+                            WormholePayload::Merkle(merkle_root) => merkle_root.root,
+                        });
+
+                    let mut verify_count = 0;
+                    for update in updates {
+                        let message_vec = Vec::from(update.message.clone());
+                        if !root.check(update.proof.clone(), &message_vec) {
+                            println!("[ERR] failed to verify update");
+                        } else {
+                            verify_count += 1;
+                        }
+                    }
+                    println!("verified {verify_count}/{} updates", updates.len());
+
+                    println!("[5/5] Post updates from AccumulatorUpdateData and use the PostedVAA on solana using pyth-solana-receiver::PostUpdates");
+                    // TODO need to figure out max number of updates that can be sent in 1 txn
+
+                    // update_bytes_len: 288 (1 price feed)
+                    let update_bytes = updates
+                        .iter()
+                        .map(|u| to_vec::<_, byteorder::BE>(u).unwrap())
+                        .collect::<Vec<_>>();
+
+                    let update_bytes_len: usize = update_bytes
+                        .iter()
+                        .map(|u| u.len())
+                        .collect::<Vec<usize>>()
+                        .iter()
+                        .sum();
+
+                    println!("update_bytes_len: {}", update_bytes_len);
+
+                    let post_updates_accounts = pyth_solana_receiver::accounts::PostUpdates {
+                        payer:         payer.pubkey(),
+                        posted_vaa:    vaa_pubkey,
+                        signature_set: *posted_vaa_data.vaa.signature_set(),
+                    }
+                    .to_account_metas(None);
+                    let post_updates_ix_data = pyth_solana_receiver::instruction::PostUpdates {
+                        price_updates: update_bytes,
```

Comment:
we should probably have a struct type for the price_updates argument that works with anchor's built in serialization/deserialization so it's easier for users to construct and work with.

Reply:
added to do for this